File: app/src/main/java/com/phz/tsnackbar/MainActivity.java
package com.phz.tsnackbar;

import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.phz.topsnackbar.TopSnackBar;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private Button btn_normal;
    private Button btn_top;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn_normal=findViewById(R.id.show_snack_bar);
        btn_top=findViewById(R.id.show_top_snack_bar);
        btn_top.setOnClickListener(this);
        btn_normal.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.show_snack_bar:
                Snackbar.make(view,"我是正常Snackbar", Snackbar.LENGTH_SHORT).show();
                break;
            case R.id.show_top_snack_bar:
                TopSnackBar.make(view,"我是顶部Snackbar", TopSnackBar.LENGTH_SHORT,TopSnackBar.STYLE_WARNING).show();
                break;
        }
    }
}

File: app/src/main/java/com/phz/tsnackbar/MainActivity.kt
package com.phz.tsnackbar

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.google.android.material.snackbar.Snackbar
import com.phz.topsnackbar.TopSnackbar
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        bt_tsnackbar_one.setOnClickListener {
            TopSnackbar.make(it,getString(R.string.my_name_is_top), TopSnackbar.LENGTH_SHORT,TopSnackbar.STYLE_ERROR).show()
        }

        bt_tsnackbar_two.setOnClickListener {
            TopSnackbar.make(it,getString(R.string.my_name_is_top), TopSnackbar.LENGTH_SHORT,TopSnackbar.STYLE_WARNING).show()
        }

        bt_tsnackbar_three.setOnClickListener {
           TopSnackbar.make(it,getString(R.string.my_name_is_top), TopSnackbar.LENGTH_SHORT,TopSnackbar.STYLE_COMPLETE).show()
        }


        show_snack_bar.setOnClickListener {
            Snackbar.make(
                it,
                "我是源版SnackBar", Snackbar.LENGTH_SHORT
            ).show()
        }
    }
}